src/services/phar/InstallService.php

```diff
@@ -80,11 +80,14 @@ private function getInstalledVersionConstraint(VersionConstraint $requestedVersi
             return $requestedVersionConstraint;
         }
 
+        /** @var int $major */
+        $major = $installedVersion->getMajor()->getValue();
+
         return new AndVersionConstraintGroup(
             sprintf('^%s', $installedVersion->getVersionString()),
             [
                 new GreaterThanOrEqualToVersionConstraint($installedVersion->getVersionString(), $installedVersion),
-                new SpecificMajorVersionConstraint($installedVersion->getVersionString(), $installedVersion->getMajor()->getValue())
+                new SpecificMajorVersionConstraint($installedVersion->getVersionString(), $major)
```

Comment:
How is this change making anything better? Doesn't it just hide the problem? Why do we do this?

Reply:
I understand that `getValue` is nullable. Maybe we should fix that in `phar-io/version` because it doesn't make sense..

Reply:
I agree that my change just hide the problem, so I've changed it a bit. That way the code is more reliable and it doesn't require to change the dependency.